### PR TITLE
[IMP] im_livechat: improve livechat session history search

### PR DESCRIPTION
**Pull request by tsm-odoo**
This PR improves the search on livechat session's history:
- Rename "Name" into "Attendee"
- Add quick filters:
  - My sessions
  - My Team
  - Session Date
  - Ratings: Good / Bad / Unrated
  - Rename "Creation Date" into "Session Date"

At the same time, this PR fixes the livechat channel demo
data since they were not realistic:
- Older messages should have lower ids
- Operators should be members of the channel
- Rating must be the last message of the channel
- Missing `email_from` for visitor messages

part of task-3332872

### CODE REVIEW

File: addons/im_livechat/views/discuss_channel_views.xml [hunk 1]
```diff
@@ -7,11 +7,18 @@
             <field name="model">discuss.channel</field>
             <field name="arch" type="xml">
                 <search string="Search history">
-                    <field name="name"/>
+                    <field name="name" string="Attendee"/>
```
Comment (seb-odoo) [inline addons/im_livechat/views/discuss_channel_views.xml:10]: Attendees maybe? it's plural in the view itself

File: addons/im_livechat/views/discuss_channel_views.xml [hunk 2]
```diff
@@ -7,11 +7,18 @@
...
+                    <filter name="filter_session_date" date="create_date" string="Session Date"/>
+                    <separator/>
+                    <filter name="filter_session_bad_rating" domain="[('rating_ids', '!=', False), ('rating_avg', '&lt;', 2.5)]" string="Bad Ratings"/>
+                    <filter name="filter_session_good_rating" domain="[('rating_ids', '!=', False), ('rating_avg', '&gt;=', 2.5)]" string="Good Ratings"/>
+                    <filter name="fiter_session_unrated" domain="[('rating_ids', '=', False)]" string="Unrated"/>
                     <group expand="0" string="Group By...">
                         <filter name="group_by_channel" string="Channel" domain="[]" context="{'group_by':'livechat_channel_id'}"/>
                         <separator orientation="vertical"/>
-                        <filter name="group_by_month" string="Creation Date" domain="[]" context="{'group_by':'create_date:month'}"/>
+                        <filter name="group_by_month" string="Session" domain="[]" context="{'group_by':'create_date:month'}"/>
```
Comment (seb-odoo) [inline addons/im_livechat/views/discuss_channel_views.xml:21]: missing Date

File: addons/im_livechat/views/discuss_channel_views.xml [hunk 3]
```diff
@@ -7,11 +7,18 @@
...
+                    <filter name="filter_session_date" date="create_date" string="Session Date"/>
+                    <separator/>
+                    <filter name="filter_session_bad_rating" domain="[('rating_ids', '!=', False), ('rating_avg', '&lt;', 2.5)]" string="Bad Ratings"/>
+                    <filter name="filter_session_good_rating" domain="[('rating_ids', '!=', False), ('rating_avg', '&gt;=', 2.5)]" string="Good Ratings"/>
+                    <filter name="fiter_session_unrated" domain="[('rating_ids', '=', False)]" string="Unrated"/>
                     <group expand="0" string="Group By...">
                         <filter name="group_by_channel" string="Channel" domain="[]" context="{'group_by':'livechat_channel_id'}"/>
                         <separator orientation="vertical"/>
-                        <filter name="group_by_month" string="Creation Date" domain="[]" context="{'group_by':'create_date:month'}"/>
+                        <filter name="group_by_month" string="Session" domain="[]" date="create_date" context="{'group_by':'create_date:month'}"/>
```
Comment (seb-odoo) [inline addons/im_livechat/views/discuss_channel_views.xml:21]: I meant Date in string, I'm not sure what adding date attribute does.

**Review by seb-odoo — APPROVED** [review]
Commented by PM.
Ok for the most part.

Still have demo structure to cleanup ideally.

**Comment by phenix-factory** [comment]
@robodoo r+

**Comment by fhe-odoo** [comment]
@tsm-odoo In this context "Participants" makes more sense than "Attendees".

**Pull request closed without merging**